src/App.tsx

```diff
@@ -21,11 +35,17 @@ export const App: React.FC = () => {
             data-cy="perPageSelector"
             id="perPageSelector"
             className="form-control"
+            value={perPage}
+            onChange={(event => {
+              setPerPage(+event.target.value);
+              setCurrentPage(1);
+            })}
           >
-            <option value="3">3</option>
-            <option value="5">5</option>
-            <option value="10">10</option>
-            <option value="20">20</option>
+            {options.map(option => (
+              <option value={option}>
```

Comment:
Don't forget to add key attribute
https://www.developerway.com/posts/react-key-attribute